From 41a3a253d55ec8afee9756eec83511d16c37c6c2 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Sun, 29 Jul 2018 19:09:31 +0800
Subject: [PATCH 4277/5242] MLK-19083 drm/imx: lcdif: remove untrue condition
 for pixel format set

commit  c252aaad141bf5941e04177da162a3d3407ab689 from
https://source.codeaurora.org/external/imx/linux-imx.git

After supporting DISPMIX power domain, the LCDIF runtime
resume callback always write '0' to 'LCDIF_CTRL' register
which will clear previous pixel format related setting.
So the previous condition by comparing format change for
setting pixel format during plane atomic update is not
true anymore.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 5f84c69799456f28fd8182fd156e9067921e9a4e)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/lcdif/lcdif-plane.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/imx/lcdif/lcdif-plane.c b/drivers/gpu/drm/imx/lcdif/lcdif-plane.c
index 7eee2026..15b015c 100644
--- a/drivers/gpu/drm/imx/lcdif/lcdif-plane.c
+++ b/drivers/gpu/drm/imx/lcdif/lcdif-plane.c
@@ -102,7 +102,6 @@ static void lcdif_plane_atomic_update(struct drm_plane *plane,
 	struct lcdif_soc *lcdif = lcdif_plane->lcdif;
 	struct drm_plane_state *state = plane->state;
 	struct drm_framebuffer *fb = state->fb;
-	struct drm_framebuffer *old_fb = old_state->fb;
 	struct drm_gem_cma_object *gem_obj = NULL;
 	u32 fb_addr, src_off, fb_idx;
 
@@ -114,8 +113,7 @@ static void lcdif_plane_atomic_update(struct drm_plane *plane,
 	 * and the fb pixel format, since the mode set will
 	 * be done in crtc's ->enable() helper func
 	 */
-	if (plane->type == DRM_PLANE_TYPE_PRIMARY &&
-	    (!old_fb || fb->format->format != old_fb->format->format))
+	if (plane->type == DRM_PLANE_TYPE_PRIMARY)
 		lcdif_set_pix_fmt(lcdif, fb->format->format);
 
 	switch (plane->type) {
-- 
1.7.9.5

